packages/toolpad-components/src/DataGrid.tsx

```diff
@@ -256,10 +306,14 @@ const DataGridComponent = React.forwardRef(function DataGridComponent(
   );
 
   const columnsInitRef = React.useRef(false);
-  const hasColumnsDefined = columnsProp && columnsProp.length > 0;
+  const isUsingDefaultData = rows === DEFAULT_ROWS;
```

Comment:
i don't think you can compare arrays with `===`
for example: `[1,2,3] === [1,2,3]` returns `false`
you need to use some way to check for deep equality

Reply:
I thought this wouldn't work as well, but for some reason it does. I don't know yet why, so I am trying to wait for a case that breaks it

Reply:
Got it! Thanks for clearing that up